src/promptflow-core/promptflow/_core/tool_meta_generator.py

```diff
@@ -175,6 +186,34 @@ def _parse_tool_from_function(
             tool_name=f.__name__,
             error_type_and_message=error_type_and_message,
         ) from e
+    # we have to do this check here as f will be dropped in the return value
+    if not allow_any_object:
```

Comment:
It seems that we could use another function to do these.

Reply:
updated.